utils/src/rpc_client.rs

```diff
@@ -95,4 +95,8 @@ impl NodeClient for crate::Rpc {
     ) -> Result<(), Error> {
         Ok(SubspaceRpcApiClient::acknowledge_archived_segment_header(self, segment_index).await?)
     }
+
+    async fn last_segment_headers(&self, limit: u64) -> Result<Vec<Option<SegmentHeader>>, Error> {
```

Comment:
`get_segment_header_by_segment_indexes` should be changed to return a segment header of the "genesis" segment to be similar to the current implementation. Currently, it returns `Vec::new` for the "last segments" request if `max_segment_index` returns `None` which differs from the farmer's implementation.

Reply:
Alternatively, it could be implemented similar to the current farmer's implementation (proxying requests to node)

Reply:
@shamil-gadelshin this is not used by the farmer. It is only there to satisfy the `NodeClient` trait.

Reply:
@ParthDesai last version of farmer does use it or else it most likely wouldn't be in the interface

Reply:
> @shamil-gadelshin this is not used by the farmer. It is only there to satisfy the `NodeClient` trait.

Indeed, however, `get_segment_header_by_segment_indexes` function is implemented incorrectly now. The correct implementation would either use or change the algorithm.

Reply:
@nazar-pc Yes. What I meant is the farmer inside subspace-sdk does not use it at least not directly. We use segment header store in `get_segment_header_by_segment_indexes`

Reply:
We might want to split `NodeClient` into two traits then

Reply:
@nazar-pc But, do we want to divide it into two traits just for SDK?